pkg/cmd/gist/edit/edit.go

```diff
@@ -147,10 +148,19 @@ func editRun(opts *EditOptions) error {
 		return errors.New("you do not own this gist")
 	}
 
+	// Transform our gist into the schema that the update endpoint expects
+	gistToUpdate := gistToUpdate{
+		id:          gist.ID,
+		Description: gist.Description,
+		Files: maps.Map(gist.Files, func(f *shared.GistFile) *gistFileToUpdate {
+			return &gistFileToUpdate{Content: f.Content, Filename: f.Filename}
+		}),
+	}
+
```

Comment:
The non-functional alternative to this looks like:

```go
files := make(map[string]*updatedGistFile, len(gist.Files))
for filename, file := range gist.Files {
	files[filename] = &updatedGistFile{
		Content:  file.Content,
		Filename: file.Filename,
	}
}
```

This isn't too bad.

If we move the transformation into the `updateGist` function then we also need to handle `nil`:

```go
files := make(map[string]*updatedGistFile, len(gist.Files))
for filename, file := range gist.Files {
	if file == nil {
		files[filename] = nil
	} else {
		files[filename] = &updatedGistFile{
			Content:  file.Content,
			Filename: file.Filename,
		}
	}
}
```

Reply:
Like I said, I feel 2/10 strongly about this. Looking at it again I feel like the first snippet is preferable to introducing the `Map` function but since I have mentioned it already above, I'll leave it as is to hear opinions.

Reply:
I also don't have super strong feelings here. I lean towards the non-functional code block as well, though, instead of introducing `Map`.

I think doing the transform earlier than later is wise since in a linear reading of the code path you learn sooner what the final shape looks like and can plan accordingly.